utils/httpsignature/personalsign.go

```diff
@@ -0,0 +1,42 @@
+package httpsignature
+
+import (
+	"crypto"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethc "github.com/ethereum/go-ethereum/crypto"
+)
+
+// EthAddress - the ethereum address hex repr
+type EthAddress string
+
+// Verify the signature sig for message based on eth address
+func (key EthAddress) Verify(message, sig []byte, opts crypto.SignerOpts) (bool, error) {
+
+	// convert key to eth address
+	address := common.HexToAddress(string(key))
+
+	// re-create the message (message is just the built http signing string)
+	m := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(message), message)
```

Comment:
Could we use [Accounts.TextHash()](https://github.com/ethereum/go-ethereum/blob/b868ca179092053365c5c6fc7cd57021af9d025e/accounts/accounts.go#L176) helper here instead of redefining the logic again? It uses `sha3.NewLegacyKeccak256()` instead of `ethc.Keccak256()`, though.

EDIT: `ethc.Keccak256()` is a [wrapper of `sha3.NewLegacyKeccak256()`](https://github.com/brave-intl/bat-go/blob/0c23ff9dde4851995769e06446e1c264c43cf17d/utils/altcurrency/altcurrency.go#L132). I'd use the `Accounts.TextHash()` helper here.